### DISCUSSION TITLE
卸载面具、安装Apatch后看到系统模块有Shamiko，正常吗？

### DISCUSSION THREAD

**Discussion by xlucn** [bmax121/APatch#613] [Q&A] [answered]
我刚刚直接卸载了面具（在面具的app内完全卸载，不过没有事先禁用或卸载面具模块）。

之后，我使用干净的boot镜像，按照官方文档修补后安装了Apatch。

现在，我看到“系统模块”页面有Shamiko v1.0.1，处于开启状态，并且无法卸载。

请教：上述现象是正常的吗？不正常的话，是为什么，如何清理？

谢谢！

---

更新：

现在装了ZygiskNext，原来的Shamiko模块就看不到了。。。这就算无事发生吗？

**Comment by ghost** [comment, marked as answer]
mt管理器 /data/adb/modules 找到shamiko删了就ok。没有就是安装zygisknext自动卸载了，那么就不用管了。
Magisk转KSU/APatch之前，建议先把所有闭源模块全部删除(什么shamiko，zygisknext v4-0.9.2以后的所有版本，Tricky Store 1.0.3以后所有版本)。鬼知道里面有没有炸弹。
确定安装好apatch之后，如果需要lsp，可以把zygisknext装回去。

**Reply by xlucn** [reply]
看了一下，`/data/adb/modules`里面没有shamiko了，那估计就是被自动卸载了。谢谢解答！
